### ISSUE TITLE
HTML renderer produces an exception and stops rendering when receiving an unordered rect

### ISSUE THREAD

**Issue by flar** [flutter/flutter#140496]
The legacy behavior of drawRect, drawOval and clipRect has been established by the implementation of Skia which automatically sorts their rects before executing the operation. As we shift our implementations we should verify this behavior at the Flutter framework level.

The problem was initially observed in https://github.com/flutter/flutter/issues/128159 when the engine started using the DisplayList to record operations and fixed for drawRect and drawOval in https://github.com/flutter/engine/pull/42556.

Writing a test case for more coverage of these instances of accepting unordered rects pointed out that we have problems with this in the latest engine as reported in https://github.com/flutter/flutter/issues/140490. The native platform code is being patched in https://github.com/flutter/engine/pull/49309, but there is still a problem in the HTML renderer which stops rendering and outputs an exception:

```
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following DomException object was thrown during a scheduler callback:
  IndexSizeError: Failed to execute 'ellipse' on 'CanvasRenderingContext2D': The major-axis radius
provided (-15) is negative.
```

The canvaskit renderer appears to pass just fine, but it should probably have a test to verify the behavior as well.

**Comment by jonahwilliams** [comment]
Moving this to team-web

**Comment by yjbanov** [comment]
Would this be a sufficient test case that we'll have to pass?

https://github.com/flutter/engine/pull/49309/files#diff-1e947a873a5774a3a1b4c8222ffdbfe01170c3ea7104892cb95347b271838ceb

**Comment by flar** [comment]
> Would this be a sufficient test case that we'll have to pass?
> 
> https://github.com/flutter/engine/pull/49309/files#diff-1e947a873a5774a3a1b4c8222ffdbfe01170c3ea7104892cb95347b271838ceb

Yes, that test should draw rects with all forms of "unordering" (LR reversed, TB reversed, and all 4 reversed) using all of the rendering operations that have historically auto-ordered the rect (formerly indirectly via Skia, but now we enforce this in ui.Canvas)
